### PR TITLE
Permissions warnings / info tooltips / confirmation modals

### PR DESCRIPTION
**Pull request by tlrobinson**
Adds info tooltips for special groups and warning tooltips if All Users group has greater permissions than another group

Partially implements #3552

### CODE REVIEW

**Comment by tlrobinson** [comment]
screenshot 2016-12-07 13 51 37
screenshot 2016-12-07 13 51 40

**Comment by mazameli** [comment]
This definitely feels like an improvement to me. 👍

**Comment by tlrobinson** [comment]
I added modals as well.

@mazameli You should probably hammer on it for a few minutes.

@metabase/core-developers Anyone else want to code review?

File: frontend/src/metabase/admin/permissions/components/PermissionsGrid.jsx [hunk 1]
```diff
@@ -210,9 +231,14 @@ class GroupPermissionCell extends Component {
...
                             }
-                            let confirmText = permission.confirm && permission.confirm(group.id, entity.id, value);
-                            if (confirmText) {
-                                this.setState({ confirmText, confirmAction });
+                            let confirmations = permission.confirm && permission.confirm(group.id, entity.id, value);
+                            if (Array.isArray(confirmations)) {
+                                confirmations = confirmations.filter(c => c);
+                            } else if (confirmations) {
+                                confirmations = [confirmations];
+                            }
```
Comment (salsakran) [inline frontend/src/metabase/admin/permissions/components/PermissionsGrid.jsx]: Re style -- Why can confirmations be an array or a single entry? This seems more error prone than having a single type for confirmations.

**Review by salsakran — APPROVED** [review]
seems reasonable, though I'm not a fan of the way "confirmations" is treated here, as well as similar patterns across the code base. 

while it's more convenient for the caller, it introduces more surface area for bugs to have the same variable take on different types/shapes.

**Comment by tlrobinson** [comment]
Ok should be ready to go.

**Comment by salsakran** [comment]
👍

**Pull request merged**